site/app/controllers/admin/ReportController.php

```diff
@@ -81,6 +81,18 @@ public function generateGradeSummaries() {
             $this->saveUserToFile($base_path, $a, $b, $c, $d);
             return null;
         });
+
+
+        if ($this->core->getConfig()->isPollsEnabled()) {
```

Comment:
I would be in favor of moving this before we generate the grade summaries themselves. That way, if we have `is_writable` failures, we hit those first before attempting to write out any data. My general preference on error handling is to cover as much up-front (especially these sort of static checks) so that we don't have to give return messages to the user like "Generated grade summaries, but failed to generate poll summaries" or whatever.